hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java

```diff
@@ -333,6 +358,16 @@ public static DatanodeDetails getFromProtoBuf(
    * @return HddsProtos.DatanodeDetailsProto
    */
   public HddsProtos.DatanodeDetailsProto getProtoBufMessage() {
+    return toProto(CURRENT_VERSION);
+  }
+
+  public HddsProtos.DatanodeDetailsProto toProto(int clientVersion) {
+    return toProtoBuilder(clientVersion).build();
+  }
+
+  public HddsProtos.DatanodeDetailsProto.Builder toProtoBuilder(
+      int clientVersion) {
```

Comment:
Can we embed the version info inside the PORT Enum without passing clientVersion as parameter?
 
This way, we can make the the handling of the version specific deserialization easier for future changes.

Reply:
Thanks @xiaoyuyao for the review.  I'm not sure I understand the proposal.  Can you please clarify?

We can embed the minimum version required for handling each port in the enum constants (eg. `0` for existing ports and `1` for the new replication port).  But don't we still need the client version to be able to decide whether to include it in the proto message or not?

Reply:
Yes. We still need the client version. But that can be a static helper without many interface changes to introduce extra parameter.

Reply:
> We still need the client version. But that can be a static helper

This conversion must be done on the server-side, and server can have clients of various versions.  How would a static helper work?